src/Build/BuildCheck/Infrastructure/AnalysisContext/AnalysisLoggingContext.cs

```diff
@@ -39,4 +41,25 @@ public void DispatchAsCommentFromText(MessageImportance importance, string messa
     public void DispatchAsErrorFromText(string? subcategoryResourceName, string? errorCode, string? helpKeyword, BuildEventFileInfo file, string message)
         => loggingService
             .LogErrorFromText(eventContext, subcategoryResourceName, errorCode, helpKeyword, file, message);
+
+    private bool ShouldUpdateProjectFileMap(BuildEventArgs buildEvent) => buildEvent.BuildEventContext != null &&
+               buildEvent.BuildEventContext.ProjectContextId != BuildEventContext.InvalidProjectContextId &&
+               !loggingService.ProjectFileMap.ContainsKey(buildEvent.BuildEventContext.ProjectContextId);
+
+    private void UpdateProjectFileMap(BuildEventArgs buildEvent)
+    {
+        string file = GetFileFromBuildEvent(buildEvent);
+        if (!string.IsNullOrEmpty(file))
+        {
+            loggingService.ProjectFileMap[buildEvent.BuildEventContext!.ProjectContextId] = file;
+        }
+    }
+
+    private string GetFileFromBuildEvent(BuildEventArgs buildEvent) => buildEvent switch
```

Comment:
because project file can be defined only in these events (and inherited from them) https://github.com/dotnet/msbuild/blob/6bc91d5e2d3d8a199fdbe367ed015b55daf57046/src/Build/BackEnd/Components/Logging/LoggingService.cs#L1198